.travis.yml

```diff
@@ -5,39 +5,41 @@ env:
 
+services:
+  - docker
+
 # todo: remove this once travis fixes their upstream issue (travis-ci/travis-ci#8836)
 sudo: required
-addons:
-  chrome: stable
 
-script:
-  # Bazel build & test everything. Can't use `bazel build ...` yet, because
-  # javaharness doesn't build on Travis (needs working Android SDK).
-  - >
-    bazel build --noshow_progress --noshow_loading_progress
-    //particles/...
-    //src/...
-    //src_kt/...
-  - >
-    bazel test --noshow_progress --noshow_loading_progress --test_output=errors
-    //particles/...
-    //src/...
-    //src_kt/...
-  - npm run test-with-start
-  - npm run build:rollup
-  - npm --prefix=server test
+# Bazel build & test everything. Can't use `bazel build ...` yet, because
+# javaharness doesn't build on Travis (needs working Android SDK).
+jobs:
+  include:
+    - stage: build
+      script: docker run -v $(pwd):/usr/src/app/ -v /home/travis/disk-cache:/disk-cache/ alxrsngrtn/arcs-travis-build:latest /bin/sh -c "cd /usr/src/app; bazel build --disk_cache=/disk-cache/ //particles/... //src/... //src_kt/... --verbose_failures"
+    - stage: test
+      script: docker run -v $(pwd):/usr/src/app/ -v /home/travis/disk-cache:/disk-cache/ alxrsngrtn/arcs-travis-build:latest /bin/sh -c "cd /usr/src/app; bazel test --disk_cache=/disk-cache/ //particles/..."
+      name: "Bazel Test: //particles/..."
+    - script: docker run -v $(pwd):/usr/src/app/ -v /home/travis/disk-cache:/disk-cache/ alxrsngrtn/arcs-travis-build:latest/bin/sh -c "cd /usr/src/app; bazel test --disk_cache=/disk-cache/ //src/..."
+      name: "Bazel Test: //src/..."
+    - script: docker run -v $(pwd):/usr/src/app/ -v /home/travis/disk-cache:/disk-cache/ alxrsngrtn/arcs-travis-build:latest /bin/sh -c "cd /usr/src/app; bazel test --disk_cache=/disk-cache/ //src_kt/..."
+      name: "Bazel Test: //src_kt/..."
+    - script: docker run -v $(pwd):/usr/src/app/ alxrsngrtn/arcs-travis-build:latest /bin/sh -c "cd /usr/src/app; npm run test-no-web"
+      name: "Node Test: Unit"
+    - script: docker run -v $(pwd):/usr/src/app/ alxrsngrtn/arcs-travis-build:latest /bin/sh -c "cd /usr/src/app; npm run test-shells"
+      name: "Node Test: Shells"
 
 cache:
   npm: true
   directories:
-    - /home/travis/.cache/bazel/
+    - /home/travis/disk-cache/
 
-# Thinking about adding install steps? Why not modify the tools/setup script instead?
-# The list command dumps the top-level packages with their versions, in case we need that for debugging.
-install:
-  - ./tools/setup
-  - npm list --depth=0
+before_install:
+  - docker pull alxrsngrtn/arcs-travis-build:latest
```

Comment:
Can you add a comment linking to where this is defined? I don't know where to look to check it out

Reply:
Generally, docker pull comes from docker hub. The URL is derived from the name of the image.

I'll add The URL tomorrow.